ruby/examples/app/controllers/user_registration_controller.rb

```diff
@@ -0,0 +1,31 @@
+class UserRegistrationController < ApplicationController
+  def new
+    @user = UserRegistration.new
+  end
+
+  def create
+    request = RegisterUser::Request.new(*user_params)
+    response = RegisterUser.new.register(request)
```

Comment:
The constructor defaults break the 

## Single-Responsibility Principle

* Because the use cases constructing their concrete dependencies are performing the responsibility of **Main**: The responsibility of knowing about what depends on what. 

What happens if the gateway requires a dependency, but not the same one for all it's clients? You've now got a very **viscous** and **rigid** system

You'll be tempted to add a line in the constructor if gateway.nil? then do some code to grab the gateway's dependency from elsewhere.

And if a use case's dependency needs to be computed based on the request, now, two classes need to change - the controller AND the use case

## Open/Closed principle

* to change the behaviour of this class used in its default form for all clients you have to modify the use case class to change its default constructor behaviour 

In a system where many default constructors are using a single Gateway, this can lead to **rigidity** and **viscosity**

## Dependency Inversion Principle

* Makes the class **immobile** to systems that need the use cases, but not the default dependency